### DISCUSSION TITLE
Inverse kinematics calculation time

### DISCUSSION THREAD

**Discussion by bhtxy0525** [stack-of-tasks/pinocchio#2036] [Q&A]
@jcarpent  Hi, sorry to disturb you.
Have your group tested the calculation time of the inverse kinematics for the manipulator robots? I didn't see any relevant data in the paper （paper title: The Pinocchio C++ library – A fast and flexible implementation of rigid body dynamics algorithms and their analytical derivatives）

I have run the inverse kinematics example (C++) on the ubuntu22.04 system, and the single calculation time is about 1.2 μs (excluding the time of model import).  Is the calculation time right?  According to my knowledge,  the damped pseudo-inverse-based method requires significant time to calculate iteratively.

**Comment by jcarpent** [comment]
Which code to you refer to?

**Comment by bhtxy0525** [comment]
C++ version
https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/md_doc_b-examples_d-inverse-kinematics.html#autotoc_md45

**Comment by jcarpent** [comment]
The timings seem good, notably if you are working with a manipulator.
